flink-filesystems/flink-s3-fs-presto/pom.xml

```diff
@@ -342,6 +338,130 @@ under the License.
+										</relocation>
+										<!-- relocate S3 presto and dependencies -->
+										<relocation>
+											<pattern>com.facebook</pattern>
+											<shadedPattern>org.apache.flink.fs.s3presto.shaded.com.facebook</shadedPattern>
+										</relocation>
+										<relocation>
+											<pattern>com.fasterxml</pattern>
+											<shadedPattern>org.apache.flink.fs.s3presto.shaded.com.fasterxml</shadedPattern>
+										</relocation>
+										<relocation>
+											<pattern>io.airlift</pattern>
+											<shadedPattern>org.apache.flink.fs.s3presto.shaded.io.airlift</shadedPattern>
+										</relocation>
+										<relocation>
+											<pattern>javax.xml.bind</pattern>
+											<shadedPattern>org.apache.flink.fs.s3presto.shaded.javax.xml.bind</shadedPattern>
+										</relocation>
+										<relocation>
+											<pattern>org.HdrHistogram</pattern>
+											<shadedPattern>org.apache.flink.fs.s3presto.shaded.org.HdrHistogram</shadedPattern>
+										</relocation>
+										<relocation>
+											<pattern>org.joda</pattern>
+											<shadedPattern>org.apache.flink.fs.s3presto.shaded.org.joda</shadedPattern>
+										</relocation>
+										<relocation>
+											<pattern>org.weakref</pattern>
+											<shadedPattern>org.apache.flink.fs.s3presto.shaded.org.weakref</shadedPattern>
+										</relocation>
+										<relocation>
+											<pattern>org.openjdk</pattern>
+											<shadedPattern>org.apache.flink.fs.s3presto.shaded.org.openjdk</shadedPattern>
+										</relocation>
+										<relocation>
+											<pattern>com.google</pattern>
+											<shadedPattern>org.apache.flink.fs.s3presto.shaded.com.google</shadedPattern>
+										</relocation>
+
+										<!-- shade Flink's Hadoop FS utility classes -->
+										<relocation>
+											<pattern>org.apache.flink.runtime.util</pattern>
+											<shadedPattern>org.apache.flink.fs.s3presto.common</shadedPattern>
+										</relocation>
+									</relocations>
+									<filters>
+										<filter>
+											<artifact>*</artifact>
+											<excludes>
+												<exclude>META-INF/maven/org.weakref/**</exclude>
+												<exclude>META-INF/maven/org.hdrhistogram/**</exclude>
+												<exclude>META-INF/maven/joda-time/**</exclude>
+												<exclude>META-INF/maven/io.airlift/**</exclude>
+												<exclude>META-INF/maven/com*/**</exclude>
+												<exclude>META-INF/maven/org.apache.flink/force-shading/**</exclude>
+												<exclude>META-INF/LICENSE.txt</exclude>
+											</excludes>
+										</filter>
+										<filter>
+											<artifact>com.facebook.presto.hadoop:hadoop-apache2</artifact>
+											<includes>
+												<include>com/facebook/presto/hadoop/**</include>
+											</includes>
+										</filter>
+									</filters>
+								</configuration>
+							</execution>
+						</executions>
+					</plugin>
+				</plugins>
+			</build>
```

Comment:
Same here. I think maintaining this shading logic can be quite a pain in the ass.